### PR TITLE
[FIX] website_slides: Don't set completion_time for encrypt/corrupt PDF 

### PR DESCRIPTION
**Pull request by nboulif**
Steps to reproduce:

  - Install eLearning module
  - Go to eLearning, open course Basics of Gardening and edit it
  - Click on `How to Grow and Harvest The Best Strawberries | Basics`
  - Upload an encrypted PDF as Attachement

Issue:

  Traceback is raised :
  utils.PdfReadError("File has not been decrypted")

Cause:

  When trying to calculate `completion_time` to read pages, we try to
  get the number of pages of the pdf but an error will be raised if the
  pdf is encrypted.

Solution:

  If not possible to get number of pages, we dont set/update the
  completion_time.

opw-2612529

### CODE REVIEW

**Comment by nboulif** [comment]
@nle-odoo nearly same issue as : https://github.com/odoo/enterprise/pull/16867

**Review by nle-odoo — APPROVED** [review]
ok, but would the getNumPages already fail? but I guess this allows to get a different error

and second question, since we are just in an _onchange, if we don't have the number of page it seems to me that we could just return without showing an error to the user (it would be only useful if showing an error was useful and I'm not sure this is the case here since it's only to update the completion time)

so in summary:

should both case cause an error, or if everything works in one or two of the cases by just ignoring that we can't get the number of page, should we just not ignore it

**Comment by nboulif** [comment]
@nle-odoo Sorry for the delay
> ok, but would the getNumPages already fail? but I guess this allows to get a different error

Indeed, this was only to get different/specific error messages.

> and second question, since we are just in an _onchange, if we don't have the number of page it seems to me that we could just return without showing an error to the user (it would be only useful if showing an error was useful and I'm not sure this is the case here since it's only to update the completion time)
> 
> so in summary:
> 
> should both case cause an error, or if everything works in one or two of the cases by just ignoring that we can't get the number of page, should we just not ignore it

Mhhh, I didn't think of that... Since, we only update the completition_time, I totally agree, a simple return should do the trick (hoping that the error will be managed further in the different flows if needed).
In this case, seems better since no issue to display PDF in viewer. :+1: 

So to conclude, I left only `pdf.getNumPages()` since we don't display the error message anymore and in both cases (corrupted or encrypted) it will raise an error.

2 little note: 
- Since it will not set/update the value of completion time, the default/old value will stay (`00:00` for new upload)
- A real corrupted/encrypted PDF can still be uploaded and can raise an issue further (like in sign), but I guess not related to this issue.

Thanks!

**Review by nle-odoo — APPROVED** [review]
looks good to me

I know that in other part of the code we did "except Exception" for exactly this, but this is possibly a source of issue (if we catched another exception that we didn't want), so you could use `except PyPDF2.utils.PdfReadError`

robodoo delegate+

**Review by nle-odoo — COMMENTED** [review]
robodoo r+
